tests/unittests/core/worker/test_experiment.py

```diff
@@ -7,125 +7,59 @@
+            assert trial is not None, 'was not able to retrieve trial for test'
 
+            get_storage().set_trial_status(trial, status=new_status)
+            assert trial.status == new_status, \
+                'Trial status should have been updated locally'
 
-@pytest.fixture()
-def patch_sample2(monkeypatch):
-    """Patch ``random.sample`` to return the first one and check call."""
-    def mock_sample(a_list, should_be_one):
-        assert type(a_list) == list
-        assert len(a_list) >= 1
-        # Part of `TestReserveTrial.test_reserve_success2`
-        assert a_list[0].status == 'new'
-        assert a_list[1].status == 'new'
-        assert a_list[2].status == 'interrupted'
-        assert a_list[3].status == 'suspended'
-        assert should_be_one == 1
-        return [a_list[-1]]
-
-    monkeypatch.setattr(random, 'sample', mock_sample)
+            trial = get_storage().get_trial(trial)
+            assert trial.status == new_status, \
+                'Trial status should have been updated in the storage'
 
+    check_status_change('completed')
+    check_status_change('broken')
+    check_status_change('reserved')
+    check_status_change('interrupted')
+    check_status_change('suspended')
+    check_status_change('new')
 
-@pytest.fixture()
-def patch_sample_concurrent(monkeypatch, create_db_instance, exp_config):
-    """Patch ``random.sample`` to return the first one and check call.
 
-    The first trial is marked as new, but in DB it is reserved.
-    """
-    def mock_sample(a_list, should_be_one):
-        assert type(a_list) == list
-        assert len(a_list) >= 1
-        # Part of `TestReserveTrial.test_reserve_race_condition`
-        if len(a_list) == 4:
-            assert a_list[0].status == 'new'
-        if len(a_list) == 3:
-            assert a_list[0].status == 'new'
-        if len(a_list) == 2:
-            assert a_list[0].status == 'interrupted'
-        if len(a_list) == 1:
-            assert a_list[0].status == 'suspended'
-
-        assert should_be_one == 1
-
-        if len(a_list) > 3:
-            # Set row's status as 'reserved' just like if it was reserved by
-            # another process right after the call to orion_db.read()
-            create_db_instance.write(
-                "trials",
-                data={"status": "reserved",
-                      "heartbeat": datetime.datetime.utcnow()},
-                query={"_id": a_list[0].id})
-            trial = create_db_instance.read("trials", {"_id": a_list[0].id})
-            assert trial[0]['status'] == 'reserved'
-
-        return [a_list[0]]
-
-    monkeypatch.setattr(random, 'sample', mock_sample)
+def test_change_status_failed_update(exp_config_file):
+    """Successfully find new trials in db and reserve one at 'random'."""
+    with OrionState(from_yaml=exp_config_file) as cfg:
+        trial = cfg.get_trial(0)
+        assert trial is not None, 'Was not able to retrieve trial for test'
 
+        try:
```

Comment:
Better practice is to use with `pytests.raises(FailedUpdate) as exc:` and then verify that `str(exc.value)` has the proper message.